services/brig/test/integration/API/Provider.hs

```diff
@@ -346,59 +346,39 @@ testListServicesByTag config db brig = do
-        let _names = map serviceProfileName _ls
-        liftIO $ assertEqual (printf "names (%d)" i) (drop i names) _names
-
-    -- Chosen prefixes
-
-    _ls <- getPage uid _tags (Just (mkName uniq "Bjø")) 17
-    let _names = map serviceProfileName _ls
-    liftIO $ assertEqual "names (>= Bjø)" (select (> 3) names) _names
-
-    _ls <- getPage uid _tags (Just (mkName uniq "chris")) 16
-    let _names = map serviceProfileName _ls
-    liftIO $ assertEqual "names (>= chris)" (select (> 4) names) _names
+    -- List services with different start names, that all start with the
+    -- same prefix.
+    searchAndAssert uid _tags (Name uniq) 20 names
 
-    -- Multiple tags (conjunction)
+    -- Search by exact name
+    forM_ names $ \n -> searchAndAssert uid _tags n 10 [n]
 
-    let _tags = matchAll (match SocialTag .&&. match QuizTag)
-    _ls <- getPage uid _tags (Just (mkName uniq "Alp")) 10
-    let _names = map serviceProfileName _ls
-    liftIO $ assertEqual "names (social*quiz)" (select odd names) _names
-
-    let _tags = matchAll (match MusicTag .&&. match SocialTag .&&. match LifestyleTag)
-    _ls <- getPage uid _tags (Just (mkName uniq "A")) 10
-    let _names = map serviceProfileName _ls
-    liftIO $ assertEqual "names (music*social*lifestyle)" (select even names) _names
+    -- Chosen prefixes
 
-    -- Multiple tags (disjunction + conjunction)
+    -- Only Bjørn should be returned
+    let _search = mkName uniq "Bjø"
+    searchAndAssert uid _tags _search 10 (select _search names)
 
-    let _tags = match1 MusicTag .||. matchAll (match BusinessTag .&&. match QuizTag)
-    _ls <- getPage uid _tags (Just (mkName uniq "A")) 12
-    let _names = map serviceProfileName _ls
-    let _idx i = i `elem` [1,7] || even i
-    liftIO $ assertEqual "names (music+business*quiz)" (select _idx names) _names
+    -- Both Bjørn and Bjorn should be returned
+    let _search = mkName uniq "Bj"
+    searchAndAssert uid _tags _search 10 (select _search names)
 
-    let _tags = matchAll (match QuizTag .&&. match BusinessTag) .||.
-                matchAll (match QuizTag .&&. match TravelTag)   .||.
-                matchAll (match QuizTag .&&. match WeatherTag)
-    _ls <- getPage uid _tags (Just (mkName uniq "A")) 10
-    let _names = map serviceProfileName _ls
-    liftIO $ assertEqual "names (quiz*(business+travel+weather))" (select odd names) _names
+    -- CHRISTMAS should be returned
+    let _search = mkName uniq "chris"
+    searchAndAssert uid _tags _search 10 (select _search names)
   where
     getPage uid tag start size = do
         rs <- listServiceProfilesByTag brig uid tag start size <!!
             const 200 === statusCode
         let Just ls = serviceProfilePageResults <$> decodeBody rs
         return ls
 
+    searchAndAssert uid tags qry size expects = do
+        _ls <- getPage uid tags (Just qry) size
+        liftIO $ assertEqual (printf "size: %s" $ show qry) (length expects) (length _ls)
```

Comment:
not sure about the use of printf here. Can we use something ala `"size" ++ show qry` ?

Reply:
👍